tests/sentry/integrations/github/test_webhooks.py

```diff
@@ -134,6 +165,54 @@ def test_simple(self, mock_get_jwt):
         assert commit.author.external_id is None
         assert commit.date_added == datetime(2015, 5, 5, 23, 40, 15, tzinfo=timezone.utc)
 
+    @patch("sentry.integrations.github.client.get_jwt")
+    def test_auto_linking_missing_feature_flag(self, mock_get_jwt):
+        mock_get_jwt.return_value = ""
+
+        project = self.project  # force creation
+
+        self._setup_repo_test(project)
+
+        repos = Repository.objects.all()
+        assert len(repos) == 0
+
+    @with_feature("organizations:auto-repo-linking")
+    @patch("sentry.integrations.github.client.get_jwt")
+    def test_creates_missing_repo(self, mock_get_jwt):
+        mock_get_jwt.return_value = ""
+
+        project = self.project  # force creation
+
+        self._setup_repo_test(project)
+
+        repos = Repository.objects.all()
+        assert len(repos) == 1
+        assert repos[0].organization_id == project.organization.id
+        assert repos[0].external_id == "35129377"
+        assert repos[0].provider == "integrations:github"
+        assert repos[0].name == "baxterthehacker/public-repo"
+
+    @with_feature("organizations:auto-repo-linking")
+    @patch("sentry.integrations.github.client.get_jwt")
+    def test_ignores_hidden_repo(self, mock_get_jwt):
+        mock_get_jwt.return_value = ""
+
+        project = self.project  # force creation
+
+        repo = Repository.objects.create(
```

Comment:
nit: use the testutil function: `self.create_repo`

Reply:
i can't set the `status` or `external_id` with the factory function. should i modify the factory function?

Reply:
you can modify the created object after it's been created
```
repo = self.create_repo()
repo.status=...
repo.save()
```